src/field.rs

```diff
@@ -34,6 +38,11 @@
 //! res = a * b; // Performs a * b (mod l).
 //! res = a.square(); // Performs a^2 (mod l).
 //! res = -&a; // Performs Negation over the modulo l.
+//! res = a.pow(&b); // Performs Modular exponentiation.
+//! res = a.mod_sqrt(Choice::from(1u8)).unwrap(); //Performs
```

Comment:
The same applies here but on a far smaller scale. I can add a thing or two from the outline you've given - which is very detailed.